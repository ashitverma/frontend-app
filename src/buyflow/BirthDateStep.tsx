import React, { useState } from "react";

interface BirthDateStepProps {
  cb: (field: string, value: string) => void;
}

const BirthDateStep: React.FC<BirthDateStepProps> = (props) => {
  const [birthDate, setBirthDate] = useState("");
  console.log(new Date().toISOString().split("T")[0]);
  return (
    <>
      <div>
        <label htmlFor="birthDate">Birth Date : </label>
        <input
          type="date"
          onChange={({ target: { value } }) => {
            setBirthDate(value);
          }}
          value={birthDate}
          max={new Date().toISOString().split("T")[0]}
        ></input>
      </div>
      <button onClick={() => props.cb("birthDate", birthDate)}>Next</button>
    </>
  );
};

export default BirthDateStep;
