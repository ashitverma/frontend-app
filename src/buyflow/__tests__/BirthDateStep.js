import { render } from "@testing-library/react"; // (or /dom, /vue, ...)
import BirthDateStep from "../BirthDateStep";

test("renders a birthDate step input field", () => {
  const queries = render(<BirthDateStep />);
  expect(queries.getByText(/birth date/i)).toBeInTheDocument();
});

it("should take a snapshot", () => {
  const { asFragment } = render(<BirthDateStep />);

  expect(asFragment(<BirthDateStep />)).toMatchSnapshot();
});
