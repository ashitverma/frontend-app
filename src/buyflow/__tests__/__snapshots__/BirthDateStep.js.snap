// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      for="birthDate"
    >
      Birth Date:
    </label>
    <input
      type="date"
      value=""
    />
  </div>
  <button>
    Next
  </button>
</DocumentFragment>
`;
